api/src/main/kotlin/org/jellyfin/apiclient/api/client/sockets/WebSocketApi.kt

```diff
@@ -81,41 +81,45 @@ class WebSocketApi(
 		.catch { it.printStackTrace() }
 		.collect()
 
-	private fun ensureConnected() {
+	private suspend fun ensureConnected() {
+		logger.debug("Validating connection")
+
 		if (socketJob == null) reconnect()
 	}
 
-	fun reconnect() {
+	suspend fun reconnect() {
 		// Get base url and access token and check if they're set
 		val baseUrl = checkNotNull(api.baseUrl)
 		val accessToken = checkNotNull(api.accessToken)
 
-		println("RECONNECT")
+		logger.debug("Reconnect requested")
 
 		// Close existing socket
 		socketJob?.cancel()
 
 		// Open socket
-		socketJob = GlobalScope.launch(Dispatchers.IO) {
-			// Create web socket connection
-			api.client.ws({
-				url {
-					// Create from base URL
-					takeFrom(baseUrl.replace("^http".toRegex(), "ws"))
-
-					// Assign path making sure to remove duplicated slashes between the base and appended path
-					encodedPath = "${encodedPath.trimEnd('/')}/socket"
-
-					// Add required parameters
-					parameter("api_key", accessToken)
-					parameter("deviceId", api.deviceInfo.id)
+		withContext(Dispatchers.Unconfined) {
+			socketJob = launch {
```

Comment:
Not 100% sure, but in this case it might even work to `launch(Dispatchers.Unconfined) {}`.

Reply:
The launch function wasn't available unless I used withContext first